11-Backbonejs/04-Backbone-Models.md

```diff
@@ -283,6 +283,23 @@ Good catch, hypothetical student! Ideally we shouldn't be calling `render()` in
 
 The reason is, if you're a view, you have no guarantee that all changes to the model will come through you. Maybe the model comes from a database server and is updated periodically. Maybe there are multiple views for the same model, and when one changes the others should pick it up. Model events can be an effective solution to all of these problems.
 
+### Re-Delegating DOM Events
+There's a bug in our application. Marking a task complete works fine, and adding a new task works fine, but once a new task has been added you can't mark _any_ task complete. Weird!
+
+The reason has to do with how DOM event listeners work. Remember that our `completeHandler` is bound to a click event on a specific button owned by our `TaskView`. But, look at `TaskListView.render()`. The first line reads
+
+```javascript
+this.listElement.empty();
```

Comment:
I believe we could also use jQuery's `detach` function instead of `empty` and this would keep the elements for our `TaskView`s around, with their event handlers still delegated.

The code for that would be something like:
```javascript
this.listElement.children().detach();
```